Snowflake.Data/Core/Session/SessionPool.cs

```diff
@@ -167,14 +277,19 @@ private SFSession NewSession(String connectionString, SecureString password)
             }
         }
 
-        private Task<SFSession> NewSessionAsync(String connectionString, SecureString password, CancellationToken cancellationToken)
+        private Task<SFSession> NewSessionAsync(String connectionString, SecureString password, SessionCreationToken sessionCreationToken, CancellationToken cancellationToken)
         {
             s_logger.Debug("SessionPool::NewSessionAsync");
             var session = s_sessionFactory.NewSession(connectionString, password);
             return session
                 .OpenAsync(cancellationToken)
                 .ContinueWith(previousTask =>
                 {
+                    if (previousTask.IsFaulted)
+                    {
+                        _sessionCreationTokenCounter.RemoveToken(sessionCreationToken);
```

Comment:
Do we have a test to see if scenario of failure during open is handled properly in terms of pool size? (cause codecov is pointing to this line).

Reply:
Not a risk since this token will get expired nevertheless